File: server/src/routes/files.ts
import fs from "node:fs";
import path from "node:path";

import { Router } from "express";

import { settings } from "../config.js";

const router = Router();

router.get("/screenshots/:filename", (req, res) => {
  const safeName = path.basename(req.params.filename);
  if (safeName !== req.params.filename) {
    res.status(400).json({ detail: "非法文件名" });
    return;
  }

  const filePath = path.join(settings.screenshotsDir, safeName);
  if (!fs.existsSync(filePath)) {
    res.status(404).json({ detail: "截图不存在" });
    return;
  }

  res.sendFile(filePath);
});

export default router;

File: app/api/files.py
from pathlib import Path

from fastapi import APIRouter, HTTPException
from fastapi.responses import FileResponse

from app.config import settings

router = APIRouter(prefix="/api/files", tags=["files"])


@router.get("/screenshots/{filename}")
async def get_screenshot(filename: str) -> FileResponse:
    safe_name = Path(filename).name
    if safe_name != filename:
        raise HTTPException(status_code=400, detail="非法文件名")

    path = settings.screenshots_dir / safe_name
    if not path.exists():
        raise HTTPException(status_code=404, detail="截图不存在")

    return FileResponse(path)
